Deps need update to compile with GHC 8.4.4

@adinapoli: it turns out I need your help here. Could you prod this code a bit and advise me what version of packages that are compatible with GHC 8.4.4 it would best work with? If you happened to solve the problem altogether, I wouldn't mind either. :) 

Ideally I'd have these dependency bumps all the way to newest versions:

```
--- a/ridley-extras/ridley-extras.cabal
+++ b/ridley-extras/ridley-extras.cabal
@@ -23,8 +23,8 @@ library
   exposed-modules:     System.Metrics.Prometheus.Ridley.Metrics.FD
   build-depends:       base >= 4.7 && < 5,
                        text,
-                       prometheus < 0.5.0.0,
-                       shelly < 1.7.0.0,
+                       prometheus < 3,
+                       shelly < 1.9.0.0,
                        microlens,
                        ekg-prometheus-adapter < 0.2.0.0,
                        mtl,
diff --git a/ridley/ridley.cabal b/ridley/ridley.cabal
index 51b01a9..f0fcb93 100644
--- a/ridley/ridley.cabal
+++ b/ridley/ridley.cabal
@@ -32,8 +32,8 @@ library
 
   build-depends:       async < 3.0.0,
                        base >= 4.7 && < 5,
-                       containers < 0.6.0.0,
-                       katip < 0.4.0.0,
+                       containers < 0.7.0.0,
+                       katip < 0.8.0.0,
                        wai-middleware-metrics < 0.3.0.0,
                        template-haskell,
                        ekg-core,
diff --git a/ridley/src/System/Metrics/Prometheus/Ridley.hs b/ridley/src/System/Metrics/Prometheus/Ridley.hs
index 7aa70d3..b80e297 100644
--- a/ridley/src/System/Metrics/Prometheus/Ridley.hs
+++ b/ridley/src/System/Metrics/Prometheus/Ridley.hs
@@ -42,7 +42,7 @@ import           Katip
 import           Lens.Micro
 import           Network.Wai.Metrics (registerWaiMetrics)
 import           System.Metrics as EKG
-import qualified System.Metrics.Prometheus.Concurrent.Http as P
+import qualified System.Metrics.Prometheus.Http.Scrape as P
 import           System.Metrics.Prometheus.Metric.Counter (add)
 import qualified System.Metrics.Prometheus.RegistryT as P
 import           System.Metrics.Prometheus.Ridley.Metrics.CPU
```

but even when I accept the older prometheus-2.0.2 (the alternative older prometheus-0.5.0 conflicts with a lot of packages), I still get into problems (though some may be caused by too new katip), e.g.:

```
    /home/mikolaj/r/ridley/ridley/src/System/Metrics/Prometheus/Ridley.hs:137:44: error:
        • Couldn't match expected type ‘ScribeSettings’
                      with actual type ‘LogEnv -> IO LogEnv’
```

--- a/ridley-extras/ridley-extras.cabal
+++ b/ridley-extras/ridley-extras.cabal
@@ -23,8 +23,8 @@ library
   exposed-modules:     System.Metrics.Prometheus.Ridley.Metrics.FD
   build-depends:       base >= 4.7 && < 5,
                        text,
-                       prometheus < 0.5.0.0,
-                       shelly < 1.7.0.0,
+                       prometheus < 3,
+                       shelly < 1.9.0.0,
                        microlens,
                        ekg-prometheus-adapter < 0.2.0.0,
                        mtl,

--- a/ridley-extras/ridley-extras.cabal
+++ b/ridley-extras/ridley-extras.cabal
@@ -1,5 +1,5 @@
 name:                ridley-extras
-version:             0.1.0.1
+version:             0.1.0.2
 synopsis:            Handy metrics that don't belong to ridley.
 description:         See README.md
 homepage:            https://github.com/iconnect/ridley/ridley-extras#readme

--- a/ridley/ridley.cabal
+++ b/ridley/ridley.cabal
@@ -1,5 +1,5 @@
 name:                ridley
-version:             0.3.1.2
+version:             0.3.1.3
 synopsis:            Quick metrics to grow your app strong.
 description:         Please see README.md
 homepage:            https://github.com/iconnect/ridley#README

--- a/ridley/ridley.cabal
+++ b/ridley/ridley.cabal
@@ -32,8 +32,8 @@ library
 
   build-depends:       async < 3.0.0,
                        base >= 4.7 && < 5,
-                       containers < 0.6.0.0,
-                       katip < 0.4.0.0,
+                       containers < 0.7.0.0,
+                       katip < 0.8.0.0,
                        wai-middleware-metrics < 0.3.0.0,
                        template-haskell,
                        ekg-core,

--- a/ridley/src/System/Metrics/Prometheus/Ridley.hs
+++ b/ridley/src/System/Metrics/Prometheus/Ridley.hs
@@ -42,7 +42,11 @@ import           Katip
 import           Lens.Micro
 import           Network.Wai.Metrics (registerWaiMetrics)
 import           System.Metrics as EKG
+#if (MIN_VERSION_prometheus(0,5,0))
+import qualified System.Metrics.Prometheus.Http.Scrape as P
+#else
 import qualified System.Metrics.Prometheus.Concurrent.Http as P
+#endif
 import           System.Metrics.Prometheus.Metric.Counter (add)
 import qualified System.Metrics.Prometheus.RegistryT as P
 import           System.Metrics.Prometheus.Ridley.Metrics.CPU
@@ -134,7 +138,11 @@ startRidleyWithStore opts path port store = do
       le <- initLogEnv (opts ^. katipScribes . _1) "production"
 
       -- Register all the externally-passed Katip's Scribe
+#if (MIN_VERSION_katip(0,5,0))
+      le' <- foldM (\le0 (n,s) -> registerScribe n s defaultScribeSettings le0) le (opts ^. katipScribes . _2)
+#else
       let le' = List.foldl' (\le0 (n,s) -> registerScribe n s le0) le (opts ^. katipScribes . _2)
+#endif
 
       -- Start the server
       serverLoop <- async $ runRidley opts le' $ do

